doc/hardware/porting/board_porting.rst

```diff
@@ -133,57 +171,127 @@ board directory to a separate repository once it's working.)
 
   The board directory name does not need to match the name of the board.
   Multiple boards can even defined be in one directory.
-  For example, for boards with multi-core SoC, a logical board might be created
-  for each core following the naming scheme `<board>_<soc-core>`, with definitions
-  for all of these different boards defined inside the same directory. This and
-  similar schemes are common for upstream vendor boards.
 
 Your board directory should look like this:
 
 .. code-block:: none
 
    boards/<ARCH>/plank
+   ├── board.yml
    ├── board.cmake
    ├── CMakeLists.txt
    ├── doc
    │   ├── plank.png
    │   └── index.rst
-   ├── Kconfig.board
+   ├── Kconfig.plank
    ├── Kconfig.defconfig
    ├── plank_defconfig
+   ├── plank_<identifier>_defconfig
    ├── plank.dts
+   ├── plank_<identifier>.dts
+   ├── plank_<identifier>.overlay
    └── plank.yaml
 
 Replace ``plank`` with your board's name, of course.
 
 The mandatory files are:
 
-#. :file:`plank.dts`: a hardware description in :ref:`devicetree
-   <dt-guide>` format. This declares your SoC, connectors, and any
-   other hardware components such as LEDs, buttons, sensors, or communication
-   peripherals (USB, BLE controller, etc).
+#. :file:`board.yml`: a YAML file describe the highlevel meta data of the board
+   such as the board name, SoCs, and variants.
+   Cpusets for multi-core SoCs are not described in this file as they are
+   inherited from the SoC's YAML description.
+
+#. :file:`plank.dts` or :file:`plank_<identifier>.dts`: a hardware description
+   in :ref:`devicetree <dt-guide>` format. This declares your SoC, connectors,
+   and any other hardware components such as LEDs, buttons, sensors, or
+   communication peripherals (USB, BLE controller, etc).
+
+#. :file:`Kconfig.plank`: the base software configuration for selecting SoC and
+   other board and SoC related setting. No Kconfig settings outside the board
+   and SoC tree must be selected. To select general Zephyr Kconfig settings the
```

Comment:
```suggestion
   other board and SoC related settings. Kconfig settings outside the board
   and SoC tree should not be selected. To select general Zephyr Kconfig settings the
```

Reply:
*must not

Reply:
"No Kconfig settings ... shall be selected" or "Kconfig .... shall not be selected"

Reply:
Thanks for the proposal, looks better, but i'll use the `must not` variant proposed commented by @nordicjm over the `shall not`.